src/App.tsx

```diff
@@ -1,24 +1,216 @@
+      setIsRemoveError(true);
+    }
+  }, []);
+
+  const handleRemoveCompletedTodos = useCallback(async () => {
+    setIsRemCompleted(true);
+
+    try {
+      await Promise.all(
+        todos
+          .filter((todo) => todo.completed)
+          .map((todo) => client.delete(`/todos/${todo.id}`)),
+      );
+
+      setTodos((prevTodos) => prevTodos.filter((todo) => !todo.completed));
+    } catch (error) {
+      setIsRemoveError(true);
+    }
+  }, [todos]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {isTodosNoEmpty && (
+            <button
+              type="button"
+              className={classNames(
+                'todoapp__toggle-all', {
+                  active: todos.every((todo) => todo.completed),
+                },
+              )}
+            />
+          )}
+
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={query}
+              onChange={(event) => handleQueryChange(event.target.value)}
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={visibleTodos}
+          onTodoRemove={handleRemoveTodo}
+          isRemCompleted={isRemCompleted}
+        />
+        {tempTodo && (
+          <TodoItem
+            todo={tempTodo}
+            onTodoRemove={handleRemoveTodo}
+            isRemovingCompleted={isRemCompleted}
+          />
+        )}
+
+        {isTodosNoEmpty && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${itemsLeft} items left`}
+            </span>
+
+            <TodoFilter
+              filterBy={filterBy}
+              onFilterChange={handleFilterChange}
+            />
+
+            <button
+              data-cy="ClearCompletedButton"
+              type="button"
+              className="todoapp__clear-completed"
+              onClick={handleRemoveCompletedTodos}
+            >
+              Clear completed
+            </button>
+          </footer>
+        )}
+
+        <div
+          className="notification is-danger is-light has-text-weight-normal"
+          hidden={!isVisibleError}
+        >
+          <button
+            type="button"
+            className="delete"
+            onClick={handleCloseError}
+          />
+
+          {isTitleEmpty && (<strong>Title can&apos;t be empty</strong>)}
```

Comment:
just combine these errors
Make some `Enum` or `Object` with Errors & use just 1 this for it